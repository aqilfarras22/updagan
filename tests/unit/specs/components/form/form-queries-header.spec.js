import BootstrapVue from 'bootstrap-vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

import formQueriesHeaderComponent from '@/components/form/form-queries-header';

const { it, describe, expect } = global;

const testPropsData = {
  title: 'title',
  backTitle: 'backTitle',
  fakeProp: '',
};

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('components/form/form-queries-header.vue', () => {
  const createMountedWrapper = (options = {}) => shallowMount(formQueriesHeaderComponent, {
    localVue,
    propsData: { ...testPropsData },
    ...options,
  });
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.fakeProp).toBeUndefined();
    expect(wrapper.vm.title).toBe(testPropsData.title);
    expect(wrapper.vm.backTitle).toBe(testPropsData.backTitle);
  });

  it('should have method which emit "back" event by click on link: back', () => {
    const wrapper = createMountedWrapper();
    // test link
    const link = wrapper.find('.backButton');
    expect(link.is('a')).toBe(true);

    link.trigger('click');
    // assert event has been emitted
    expect(wrapper.emitted().back).toBeTruthy();
    // assert event count
    expect(wrapper.emitted().back.length).toBe(1);
  });

  it('has the expected html structure', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
