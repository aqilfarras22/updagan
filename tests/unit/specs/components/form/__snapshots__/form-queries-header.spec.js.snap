// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-queries-header.vue has the expected html structure 1`] = `
<header
  fakeprop=""
>
  <b-row-stub
    class="py-2 align-middle"
    tag="div"
  >
    <b-col-stub
      class="text"
      tag="div"
    >
      title
    </b-col-stub>
     
    <b-col-stub
      class="text text-right"
      col="true"
      lg="3"
      tag="div"
    >
      <a
        class="backButton"
        href="#"
      >
        backTitle
      </a>
    </b-col-stub>
  </b-row-stub>
</header>
`;
